### PR TITLE
Add API

### LINKED ISSUE #308: API: current status
* /api/status/
returns full status 
* /api/feature/{slug}
returns current status of a feature

### PR DESCRIPTION
**Pull request by zalun**
fixes #308

### CODE REVIEW

File: routes/helper.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+function handleErrorResponse(err, res) {
```
Comment (marco-c) [inline routes/helper.js]: Nit: directly define this in app.js since it's only used there

File: app.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+const express = require('express');
+const fs = require('fs');
+const compression = require('compression');
+const app = express();
+const notifications = require('./routes/api');
```
Comment (marco-c) [inline app.js]: Nit: call this `notificationsAPI` or something else to make it clearer.

File: server.js [hunk 3]
```diff
@@ -1,44 +1,4 @@
...
-  next();
-});
-
-app.use(compression());
-
-if (!fs.existsSync(distPublicDir)) {
-  throw new Error('Missing `dist` folder, execute `npm run build` first.');
-}
-app.use(express.static(distPublicDir));
+const app = require('./app');
```
Comment (marco-c) [inline server.js]: Why moving this to another file?

File: engine/digger.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+// return status as an object
+function getStatus(dbNumber) {
+  return setClient(dbNumber)
+  .then(() => redis.get(client, 'status'))
+  .then(status => JSON.parse(status));
+}
+
+// return status of a feature as an object
+function getFeatureStatus(slug, dbNumber) {
+  return setClient(dbNumber)
```
Comment (marco-c) [inline engine/digger.js]: Nit: Use `getStatus` instead of repeating this code.

File: routes/api.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+router.get('/api/status', (req, res) =>
+  digger.getStatus()
+  .then(status => res.json(status))
+  .catch(err => helper.handleErrorResponse(err, res))
+);
+
+router.get('/api/feature/:slug', (req, res) => {
+  const slug = req.params.slug;
+  return digger.getFeatureStatus(slug)
```
Comment (marco-c) [inline routes/api.js]: Nit: don't define `slug` and directly use `req.params.slug` here

File: tests/apiUnit.js [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+    // clean and quit database after each test
+    bdd.beforeEach(() => digger.setClient(5));
+
+    bdd.afterEach(() =>
+      digger.setClient(5)
+      .then((client) => flushQuitDB(client)));
+
+    bdd.describe('getStatus', () => {
+      bdd.it('should return null if no status in db', () =>
+        digger.getStatus(5)
```
Comment (marco-c) [inline tests/apiUnit.js]: I think you have to return the promises, no?

File: server.js (hunk 3 above)
Comment (marco-c) [inline server.js]: Ah, because you need it for tests. OK.

File: routes/helper.js (hunk 1 above)
Comment (zalun) [inline routes/helper.js]: it's gonna be used in notifications soon

File: tests/apiUnit.js (hunk 6 above)
Comment (zalun) [inline tests/apiUnit.js]: it returns - there is no "{" at the end of the previous line

File: app.js (hunk 2 above)
Comment (zalun) [inline app.js]: argh - it shouldn't be call notificatins (!) :)

File: tests/apiUnit.js (hunk 6 above)
Comment (marco-c) [inline tests/apiUnit.js]: Oops, I didn't notice you were using arrow functions :)

File: tests/apiUnit.js [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+        return digger.setClient(5)
+        .then(() => engine.saveData([testData1, testData2], 5))
+        .then(() => api()
+          .get('/api/feature/feature')
+          .send()
+        )
+        .then(response => {
+          assert.property(response.body, 'slug');
+          assert.deepEqual(response.body, testData1);
```
Comment (marco-c) [inline tests/apiUnit.js]: Nit: also check the second feature

File: tests/apiUnit.js [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+          updated: {} };
+
+        return digger.setClient(5)
+        .then(() => engine.saveData([testData1, testData2], 5))
+        .then(() => api()
+          .get('/api/feature/feature')
+          .send()
+        )
+        .then(response => {
+          assert.property(response.body, 'slug');
```
Comment (marco-c) [inline tests/apiUnit.js]: Nit: also check the value of the slug property

Comment (marco-c) [inline tests/apiUnit.js]: No need, you're using `deepEqual` later. So I guess you can remove this check then.

File: tests/config/intern-api.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+define({
```
Comment (marco-c) [inline tests/config/intern-api.js]: I'm not faimiliar with intern, but why is this file needed when you're adding `tests/apiUnit` to intern-node.js?

Comment (zalun) [inline tests/config/intern-api.js]: as I'm sick of running entire suite test (this needs to be removed though) :)
